pkg/k8s/resource_ctors.go

```diff
@@ -243,12 +258,40 @@ func EndpointsResource(lc cell.Lifecycle, cfg Config, cs client.Clientset) (reso
 	if !cs.IsEnabled() {
 		return nil, nil
 	}
-	endpointsOptsModifier, err := utils.GetServiceAndEndpointListOptionsModifier(cfg.K8sServiceProxyName)
+	endpointsOptsModifier, err := utils.GetServiceAndEndpointListOptionsModifier(cfg.K8sServiceProxyName, true)
+	if err != nil {
+		return nil, err
+	}
+
+	endpointSliceOpsModifier, err := utils.GetEndpointSliceListOptionsModifier(true)
+	if err != nil {
+		return nil, err
+	}
+	lw := &endpointsListerWatcher{
+		cs:                         cs,
+		enableK8sEndpointSlice:     cfg.EnableK8sEndpointSlice,
+		endpointsOptsModifier:      endpointsOptsModifier,
+		endpointSlicesOptsModifier: endpointSliceOpsModifier,
+	}
+	return resource.New[*Endpoints](
+		lc,
+		lw,
+		resource.WithLazyTransform(lw.getSourceObj, transformEndpoint),
+		resource.WithMetric("Endpoint"),
+		resource.WithName("endpoints"),
+	), nil
+}
+
+func EndpointsResourceOperator(lc cell.Lifecycle, cfg Config, cs client.Clientset) (resource.Resource[*Endpoints], error) {
```

Comment:
I'd suggest creating a wrapper `EndpointResourceFactory` function which takes the `excludeHeadlessServices` parameter, and returns a `func (lc cell.Lifecycle, cfg Config, cs client.Clientset) (resource.Resource[*Endpoints], error)`, to avoid duplicating the entire logic. Same for the services.